src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContextHelpers.cs

```diff
@@ -36,6 +43,14 @@ internal static void VerifyArguments(Diagnostic diagnostic, Func<Diagnostic, boo
             {
                 throw new ArgumentException(string.Format(CodeAnalysisResources.UnsupportedDiagnosticReported, diagnostic.Id), nameof(diagnostic));
             }
+
+            if (!diagnostic.Id.All(char.IsLetterOrDigit))
```

Comment:
Thanks Neal, can you point me to the language specification/piece of code in compilers where we perform these validations?